### PR TITLE
redux was added

### PR DESCRIPTION
**Pull request by bmitioglov**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -30,6 +30,8 @@
   },
   "homepage": "http://bmitioglov.github.io/learn-reactjs",
   "dependencies": {
+    "@reduxjs/toolkit": "^1.5.0",
+    "axios": "^0.21.1",
```
Comment (endiny) [inline package.json:34]: In future, I guess default `fetch` function can satisfy all your needs

File: src/components/Menu/menu.js [hunk 2]
```diff
@@ -1,14 +1,38 @@
...
+        </li>
+        <li>
+            <button onClick={fetchWithFilter('Comedy')}>Comedy</button>
+        </li>
+        <li>
+            <button onClick={fetchWithFilter('Horror')}>Horror</button>
+        </li>
+        <li>
+            <button onClick={fetchWithFilter('Crime')}>Crime</button>
+        </li>
```
Comment (endiny) [inline src/components/Menu/menu.js]: I bet that one can be rendered in a loop

File: src/components/MovieCard/moviecard.js [hunk 3]
```diff
@@ -37,7 +37,8 @@ const MovieCard = ({ title, description, year, onClick }) => {
   return (
     <div className="movie-card">
       <div className="image-container">
-        <img onClick={onClick} className="movie-image" src={movieimage} alt={title} />
+        <img onClick={onClick} className="movie-image" src={image}
+             onError={(e)=>{e.target.onerror = null; e.target.src=defaultmovieimage}}/>
```
Comment (endiny) [inline src/components/MovieCard/moviecard.js]: this logic can be extracted from the component and placed along

Comment (endiny) [reply src/components/MovieCard/moviecard.js]: `e.target.onerror = null;`
I think that one is not really necessary and also I believe handler will be set again after rerender

File: src/reducers/movieSlice.js [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+    [deleteMovie.rejected]: (state, action) => {
+      state.status = 'failed';
+      state.error = action.error.message;
+    },
+  }
+});
+
+export const { sortBy } = moviesSlice.actions;
+
+export const selectAllMovies = (state) => (state.movies.moviesData !== undefined ? state.movies.moviesData : []);
```
Comment (endiny) [inline src/reducers/movieSlice.js]: Taking in account that `moviesData` is `[]` by default and you're not assigning `undefined` anywhere, I guess that check is redundant as `moviesData` is always an array

File: src/reducers/movieSlice.js [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+
+export const selectMovieById = (state, movieId) => state.movies.moviesData.find((movie) => movie.id === movieId);
```
Comment (endiny) [inline src/reducers/movieSlice.js]: unused code?

File: src/reducers/movieSlice.js [hunk 6]
```diff
@@ -0,0 +1,107 @@
...
+function dynamicSort(property) {
+  const sortOrder = 1;
+  return function (a, b) {
+    /* next line works with strings and numbers,
+     * and you may want to customize it to your needs
+     */
+    const result = (a[property] < b[property]) ? -1 : (a[property] > b[property]) ? 1 : 0;
+    return result * sortOrder;
+  };
+}
```
Comment (endiny) [inline src/reducers/movieSlice.js]: extract to some utils

File: src/reducers/movieSlice.js [hunk 7]
```diff
@@ -0,0 +1,107 @@
...
+    [fetchMovies.pending]: (state, action) => {
+      state.status = 'loading';
+    },
+    [fetchMovies.fulfilled]: (state, action) => {
+      state.status = 'succeeded';
+      state.moviesData = action.payload.data;
+      state.total = action.payload.total;
+      state.offset = action.payload.offset;
+      state.limit = action.payload.limit;
+
```
Comment (endiny) [inline src/reducers/movieSlice.js]: extra line

File: src/reducers/movieSlice.js [hunk 8]
```diff
@@ -0,0 +1,107 @@
...
+    [updateMovie.fulfilled]: (state, action) => {
+      const objIndex = state.moviesData.findIndex((movie => movie.id === action.payload.id));
+      state.moviesData[objIndex] = action.payload;
+    },
+    [updateMovie.rejected]: (state, action) => {
+      state.status = 'failed';
+      state.error = action.error.message;
+    },
+    [deleteMovie.fulfilled]: (state, action) => {
+      const objIndex = state.moviesData.findIndex((movie => movie.id === action.payload.id));
```
Comment (endiny) [inline src/reducers/movieSlice.js]: nit: `state.moviesData.filter(({id}) => id !== action.payload.id)` looks a bit shorter

**Review by endiny — CHANGES_REQUESTED** [review]
Also please remove commented code

File: src/components/Menu/menu.js (hunk 2 above)
Comment (bmitioglov) [reply src/components/Menu/menu.js]: fixed

File: src/components/MovieCard/moviecard.js (hunk 3 above)
Comment (bmitioglov) [reply src/components/MovieCard/moviecard.js]: removed onError handler

File: src/reducers/movieSlice.js (hunk 4 above)
Comment (bmitioglov) [reply src/reducers/movieSlice.js]: fixed

File: src/reducers/movieSlice.js (hunk 5 above)
Comment (bmitioglov) [reply src/reducers/movieSlice.js]: fixed

File: src/reducers/movieSlice.js (hunk 6 above)
Comment (bmitioglov) [reply src/reducers/movieSlice.js]: fixed

File: src/reducers/movieSlice.js (hunk 7 above)
Comment (bmitioglov) [reply src/reducers/movieSlice.js]: fixed

File: src/reducers/movieSlice.js (hunk 8 above)
Comment (bmitioglov) [reply src/reducers/movieSlice.js]: fixed

**Pull request merged**